client/client.go

```diff
@@ -26,21 +26,36 @@ import (
 	"github.com/awnumar/memguard"
 	"github.com/lestrrat-go/jwx/v2/jwa"
 	"github.com/lestrrat-go/jwx/v2/jwk"
-	"github.com/lestrrat-go/jwx/v2/jws"
 
 	"github.com/openpubkey/openpubkey/gq"
 	"github.com/openpubkey/openpubkey/pktoken"
 	"github.com/openpubkey/openpubkey/pktoken/clientinstance"
 	"github.com/openpubkey/openpubkey/util"
+	"github.com/openpubkey/openpubkey/verifier"
 )
 
-// OpkClient is the OpenPubkey client
+// Interface for interacting with the OP (OpenID Provider)
+type OpenIdProvider interface {
+	RequestTokens(ctx context.Context, cicHash string) (*memguard.LockedBuffer, error)
+	Verifier() verifier.ProviderVerifier
+}
+
+type BrowserOpenIdProvider interface {
+	OpenIdProvider
+	HookHTTPSession(h http.HandlerFunc)
+}
+
+type PKTokenVerifier interface {
+	VerifyPKToken(ctx context.Context, pkt *pktoken.PKToken, extraChecks ...verifier.Check) error
+}
+
 type OpkClient struct {
-	Op     OpenIdProvider
-	cosP   *CosignerProvider
-	signer crypto.Signer
-	alg    jwa.KeyAlgorithm
-	signGQ bool // Default is false
+	Op       OpenIdProvider
+	cosP     *CosignerProvider
+	signer   crypto.Signer
+	alg      jwa.KeyAlgorithm
+	signGQ   bool // Default is false
+	verifier PKTokenVerifier
```

Comment:
The OP has all of the knowledge required to verify the OP, that's why the OP's verifier returns a `ProviderVerifier`, it does not have knowledge about cosigner verification and should not. 

The PKTokenVerifier is there to represent an entire verifier, this allows users who may have general verifiers to use that instead of something more specific to this single PK token. This is entirely for flexibility and I do not see it as something that is easily misconfigured unless someone is explicitly looking to use it.

Reply:
I agree the OP should and does not have the information to verify the cosigner. We are discussing the client here which if it is handling also getting the PK Token cosigned, should have a cosigner object which can verify the cosigned PK Token.

> This is entirely for flexibility and I do not see it as something that is easily misconfigured unless someone is explicitly looking to use it.

The only use for this seems to be to avoid changing the verification logic in the VerifierProvider in the OP? But shouldn't we be encouraging developers to add the customer verification logic in the VerifierProvider not in multi OP Verifier?

What is the usecase where it would be better for a user to put custom verification logic in the multi OP Verifier than in the VerifierProvider in the OP?